src/index.ts

```diff
@@ -44,109 +53,219 @@ export class Ollama {
 		this.fetch = f;
 	}
 
-	async tags (): Promise<Tag[]> {
-		const response = await utils.get(this.fetch, `${this.config.address}/api/tags`);
-		const json = await response.json() as TagsResponse;
-
-		return json.models.map(m => ({
-			name: m.name,
-			modifiedAt: new Date(m.modified_at),
-			size: m.size
-		}));
+    private async processStreamableRequest<T extends object>(endpoint: string, request: { stream?: boolean } & Record<string, any>): Promise<T | AsyncGenerator<T>> {
+        request.stream = request.stream ?? false;
+        const response = await utils.post(this.fetch, `${this.config.address}/api/${endpoint}`, { ...request });
+    
+        if (!response.body) {
+            throw new Error("Missing body");
+        }
+    
+        const itr = utils.parseJSON<T | ErrorResponse>(response.body);
+    
+        if (request.stream) {
+            return (async function* () {
+                for await (const message of itr) {
+                    if ('error' in message) {
+                        throw new Error(message.error);
+                    }
+                    yield message;
+                    // message will be done in the case of chat and generate
+                    // message will be success in the case of a progress response (pull, push, create)
+                    if ((message as any).done || (message as any).status === "success") {
+                        return;
+                    }
+                }
+                throw new Error("Did not receive done or success response in stream.");
+            })();
+        } else {
+            const message = await itr.next();
+            if (!message.value.done && (message.value as any).status !== "success") {
+                throw new Error("Expected a completed response.");
+            }
+            return message.value;
+        }
+    }
+
+    private async encodeImage(image: Uint8Array | string): Promise<string> {
+        if (typeof image === 'string') {
```

Comment:
Nit: might be worth aligning this more with the python library:

1. If it's a path, ready as a path
2.if it's an `Uint8Array` or `Buffer`, base64 encode it (what `fs.readFile` would return)
3. if it's already base64 encoded just accept it (I don't think the python version does this yet)

```
def _encode_image(image) -> str:
  if p := _as_path(image):
    b64 = b64encode(p.read_bytes())
  elif b := _as_bytesio(image):
    b64 = b64encode(b.read())
  else:
    raise RequestError('images must be a list of bytes, path-like objects, or file-like objects')

  return b64.decode('utf-8')
```